puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/test_foreman.groovy

```diff
@@ -0,0 +1,199 @@
+                        GEMSET = 'ruby-2.5-mysql'
+                    }
+                    stages {
+                        stage("setup-2.5-mysql") {
+                            steps {
+                                git url: 'https://github.com/theforeman/foreman', branch: foreman_branch
+                                configureRVM(env.RUBY, env.GEMSET)
+                                databaseFile(gemset(env.GEMSET), 'mysql')
+                                configureDatabase(env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("unit-tests-2.5-mysql") {
+                            steps {
+                                withRVM(['bundle exec rake jenkins:unit TESTOPTS="-v" --trace'], env.RUBY, env.GEMSET)
+                            }
+                        }
+                    }
+                    post {
+                        always {
+                            cleanup(env.RUBY, env.GEMSET)
+                            deleteDir()
+                        }
+                    }
+                }
+                stage('ruby-2.5-sqlite3') {
+                    agent { label 'fast' }
+                    environment {
+                        RUBY = '2.5'
+                        GEMSET = 'ruby-2.5-sqlite3'
+                    }
+                    stages {
+                        stage("setup-2.5-sqlite3") {
+                            steps {
+                                git url: 'https://github.com/theforeman/foreman', branch: foreman_branch
+                                configureRVM(env.RUBY, env.GEMSET)
+                                databaseFile(gemset(env.GEMSET), 'sqlite3')
+                                configureDatabase(env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("unit-tests-2.5-sqlite3") {
+                            steps {
+                                withRVM(['bundle exec rake jenkins:unit TESTOPTS="-v" --trace'], env.RUBY, env.GEMSET)
+                            }
+                        }
+                    }
+                    post {
+                        always {
+                            cleanup(env.RUBY, env.GEMSET)
+                            deleteDir()
+                        }
+                    }
+                }
+            }
+        }
+        stage('packaging') {
+            steps {
+                build(
+                    job: 'release_nightly_build_deb',
+                    propagate: false,
+                    wait: false,
+                    parameters: [
+                        string(name: 'project', value: 'foreman'),
+                        string(name: 'jenkins_job', value: env.JOB_NAME),
+                        string(name: 'jenkins_job_id', value: env.BUILD_ID)
+                    ]
+                )
+                build(
+                    job: 'release_nightly_build_rpm',
```

Comment:
We should have a dedicated release pipeline for this after JJB has applied the latest config (foreman-develop-release IIRC).

Reply:
I was wrong and that job doesn't exist. Disregard my comment.

Reply:
Will be once https://github.com/theforeman/foreman-infra/pull/923 is added,  I will update this job to reflect that.